mountpoint-s3/src/prefetch/part_queue.rs

```diff
@@ -84,13 +93,21 @@ impl<E: std::error::Error + Send + Sync> PartQueue<E> {
             Ok(part)
         }
     }
+
+    pub fn bytes_received(&self) -> usize {
+        self.bytes_received.load(Ordering::SeqCst)
+    }
 }
 
 impl<E: std::error::Error + Send + Sync> PartQueueProducer<E> {
     /// Push a new [Part] onto the back of the queue
     pub fn push(&self, part: Result<Part, PrefetchReadError<E>>) {
-        // Unbounded channel will never actually block
+        let part_size = part.as_ref().ok().map(|part| part.len());
         let send_result = self.sender.send_blocking(part);
+        if let Some(part_size) = part_size {
+            self.bytes_sent.fetch_add(part_size, Ordering::SeqCst);
+        }
+        // Unbounded channel will never actually block
         if send_result.is_err() {
             trace!("closed channel");
         }
```

Comment:
The comment refers to `send_blocking`, so reorder like this:

```suggestion
        let part_size = part.as_ref().ok().map(|part| part.len());
        // Unbounded channel will never actually block
        let send_result = self.sender.send_blocking(part);
        if send_result.is_err() {
            trace!("closed channel");
        }
        if let Some(part_size) = part_size {
            self.bytes_sent.fetch_add(part_size, Ordering::SeqCst);
        }
```